json_journal/data_analysis.py

```diff
@@ -28,18 +34,20 @@ def parent_account_balances(full_account_name_balances):
     return new_balances
 
 
-def total_profit(all_balances):
+def total_profit(full_account_name_balances):
+    '''all_balances -> dictionary of full account names and balances for each account.
+    Returns profit value'''
     expenses = 0
     income = 0
-    for account in all_balances: 
+    for account in full_account_name_balances:
         if account.startswith("expense:"):
-            expenses += all_balances[account]
+            expenses += full_account_name_balances[account]
         else:
             pass
-    for account in all_balances: 
+    for account in full_account_name_balances:
         if account.startswith("income:"):
-            income += all_balances[account]
+            income += full_account_name_balances[account]
```

Comment:
Could we use one for-loop here instead of two?

Reply:
done - might not be the most effective way to have the for loop but functionality and tests all work

Reply:
Looks fine to me

Reply:
I wouldn't want to spend tooooooo much time over-engineering a solution. Perfect is the enemy of good.